frontend/services/collectors/odiglet.go

```diff
@@ -0,0 +1,79 @@
+package collectors
+
+import (
+    "context"
+    "sort"
+    "strings"
+
+    "github.com/odigos-io/odigos/api/k8sconsts"
+    "github.com/odigos-io/odigos/frontend/graph/model"
+    "github.com/odigos-io/odigos/frontend/kube"
+    "github.com/odigos-io/odigos/frontend/services"
+    "github.com/odigos-io/odigos/k8sutils/pkg/env"
+    appsv1 "k8s.io/api/apps/v1"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func GetOdigletDaemonSetInfo(ctx context.Context) (*model.CollectorDaemonSetInfo, error) {
+    ns := env.GetCurrentNamespace()
+    name := k8sconsts.OdigletDaemonSetName
+
+    ds, err := kube.DefaultClient.AppsV1().DaemonSets(ns).Get(ctx, name, metav1.GetOptions{})
+    if err != nil { return nil, err }
+
+    status, inProgress := computeDaemonSetStatus(ds)
+    nodes := &model.NodesSummary{ Desired: int(ds.Status.DesiredNumberScheduled), Ready: int(ds.Status.NumberReady) }
+
+    result := &model.CollectorDaemonSetInfo{ Status: status, Nodes: nodes, RolloutInProgress: inProgress, ImageVersion: services.StringPtr(extractDaemonSetImageVersion(ds)), LastRolloutAt: services.StringPtr(findDaemonSetLastRolloutTime(ctx, ds)) }
+    
+	if rr := extractDaemonSetResources(ds); rr != nil { result.Resources = rr }
+    
+	return result, nil
+}
+
+func computeDaemonSetStatus(ds *appsv1.DaemonSet) (model.WorkloadStatus, bool) {
+    desired := ds.Status.DesiredNumberScheduled
+    updated := ds.Status.UpdatedNumberScheduled
+    available := ds.Status.NumberAvailable
+    ready := ds.Status.NumberReady
+    unavailable := ds.Status.NumberUnavailable
+
+    if available == 0 { return model.WorkloadStatusDown, updated < desired || available < desired }
+    if updated < desired || available < desired { if unavailable > 0 { return model.WorkloadStatusDegraded, true }; return model.WorkloadStatusUpdating, true }
+    if desired == updated && desired == available && desired == ready && unavailable == 0 { return model.WorkloadStatusHealthy, false }
+    
+	return model.WorkloadStatusUnknown, false
+}
+
+func extractDaemonSetResources(ds *appsv1.DaemonSet) *model.Resources {
+    for _, c := range ds.Spec.Template.Spec.Containers { if c.Name == k8sconsts.OdigletContainerName {
+        req := buildResourceAmounts(c.Resources.Requests)
+        lim := buildResourceAmounts(c.Resources.Limits)
+        if req == nil && lim == nil { return nil }
+        return &model.Resources{ Requests: req, Limits: lim }
+    }}
+    return nil
+}
+
+func extractDaemonSetImageVersion(ds *appsv1.DaemonSet) string {
+    for _, c := range ds.Spec.Template.Spec.Containers { if c.Name == k8sconsts.OdigletContainerName {
+        image := c.Image; if idx := strings.Index(image, "@"); idx >= 0 { image = image[:idx] }
+        parts := strings.Split(image, "/"); last := parts[len(parts)-1]
+        if colon := strings.LastIndex(last, ":"); colon >= 0 { return last[colon+1:] }
+        return last
+    }}
```

Comment:
this code is duplicate of the `Gateway`. consider making a util function and calling same logic for both cases.